src/main/scala/com.snowplowanalytics.snowplow.analytics.scalasdk/Event.scala

```diff
@@ -181,6 +180,14 @@ case class Event(app_id:                   Option[String],
       case (k, v) => Some((k, v))
     }
 
+  def orderedAtomic: List[(String, Option[Json])] =
+    Event.parser.knownKeys.flatMap(key => key.toString.substring(1) match {
```

Comment:
@rzats that's clever! I though it will be way harder. Can I propose a change here? Sorry for flip-flopping, but I just realized that if we'll have both key and value (like you did) and leave these `contexts`, `derived_contexts` and `unstruct_event` value then user will be able to decide on their own if these columns are required (very rarely we need do them), so not filtering them out gives more flexibility.

Also, what `substring(1)` does?

Reply:
The keys are actually `Symbol` instances, so they are prepended by a `'` when stringified.

Reply:
Then `.name` method must be used!